smartsim/_core/shell/shellLauncher.py

```diff
@@ -27,14 +27,19 @@
 
 from __future__ import annotations
 
-import os
+import io
+import pathlib
 import subprocess as sp
 import typing as t
 
 import psutil
 
 from smartsim._core.arguments.shell import ShellLaunchArguments
-from smartsim._core.dispatch import _EnvironMappingType, _FormatterType, dispatch
+from smartsim._core.dispatch import (
+    _EnvironMappingType,
+    _FormatterType,
+    _WorkingDirectory,
```

Comment:
If we are now exporting these alias, consider removing the leading `_`

Reply:
ah hmm so _ mean local to file?

Reply:
Generally speaking, that's the convention. A leading underscore mean "please don't touch this" or "don't worry about this, its not important"